clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -350,8 +351,12 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
                 }
 
                 /* if channel is not ready finish prepare */
-                if (channel.isConnected() && !channel.ready())
+                if (channel.isConnected() && !channel.ready()) {
                     channel.prepare();
+                    if (channel.ready()) {
+                        this.sensors.connectionPrepared.record();
```

Comment:
Hi I know the block is skipped for PLAINTEXT. 
Actually we do not care that this rate is always 0 for PLAINTEXT - calling it authenticated would work for us. 

We wanted from an ops perspective to check how many requests go wasted either because of failed TLS handshakes or failed authentications.

In fact this metric would be more useful if we add tags for the listener the network thread belongs to  https://issues.apache.org/jira/browse/KAFKA-4982

I'll copy this comment on the JIRA so it goes on the dev list.